### PR TITLE
Improve test launch

### PR DESCRIPTION
**Pull request by axuan25**
* Launch all tests through `clientTest`.
* Specify scopes manually in each test
* remove sleep(1000) !!

Tests arenow  faster and more stable!

### CODE REVIEW

**Comment by axuan25** [comment]
Want ROCKET SPEED testing ? Enable `Allow parallel run` in your Run/Debug Configuration.

**Comment by roxanemace** [comment]
Can you also please delete the `identity-android-sdk/app/src/test/java/com/reach5/identity/sdk/demo/ExampleUnitTest.ktExampleUnitTest` test?

File: app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt [hunk 1]
```diff
@@ -35,6 +35,7 @@ class MainActivityTest {
 
     private val DOMAIN = dotenv["DOMAIN"] ?: ""
     private val CLIENT_ID = dotenv["CLIENT_ID"] ?: ""
+    private val defaultSdkConfig: SdkConfig = SdkConfig(DOMAIN, CLIENT_ID)
 
     private fun getRandomSeed() = dotenv["RANDOM_SEED"]?.toInt() ?: Random.nextInt(1000)
     private val random: Random = Random(getRandomSeed())
```
[the remark this answers is not in the document]
Comment (axuan25) [reply app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt]: Used on line 810 :)

File: app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt [hunk 2]
```diff
@@ -46,24 +47,29 @@ class MainActivityTest {
     var exceptionRule: ExpectedException = ExpectedException.none()
 
     @Test
-    fun testSuccessfulReachFiveClientInstantiation() {
```
[the remark this answers is not in the document]
Comment (axuan25) [reply app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt]: It's redundant and can never fail.

File: app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt [hunk 3]
```diff
@@ -46,24 +47,29 @@ class MainActivityTest {
...
+                failure = { failWithReachFiveError(it) }
+            )
+        }
     }
 
     @Test
     fun testSuccessfulClientConfigFetch() = clientTest(initialize = false) { client ->
         val profile = aProfile()
 
+        // given an uninitialized client
```
[the remark this answers is not in the document]
Comment (axuan25) [reply app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt]: You can make a suggestion in the comment to facilitate :p

File: app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt [hunk 4]
```diff
@@ -814,74 +763,73 @@ class MainActivityTest {
...
+                failure = { failWithReachFiveError(it) }
+            )
+            else block(client)
+        }
+
+        Unit
     }
 
-    private val fullWriteScope = setOf("full_write")
+    private val fullWrite = setOf("full_write")
```
[the remark this answers is not in the document]
Comment (axuan25) [reply app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt]: Only using them as a quick shortcuts. I don't personally think they have any relevance in ReachFive

File: app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (roxanemace) [reply app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt]:
```suggestion
        // Given an uninitialized client
```

File: app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt [hunk 5]
```diff
@@ -86,248 +85,203 @@ class MainActivityTest {
...
         )
-
-        sleep(1000)
     }
 
     @Test
-    fun testSuccessfulSignupWithEmail() {
-        val client = instantiateReachFiveClient()
-
+    fun testSuccessfulSignupWithEmail() =clientTest { client ->
```
Comment (axuan25) [inline app/src/androidTest/java/com/reach5/identity/sdk/demo/MainActivityTest.kt]:
```suggestion
    fun testSuccessfulSignupWithEmail() = clientTest { client ->
```

**Pull request merged**